Fundsoft/refactor_plan-do_update.md

```diff
@@ -0,0 +1,337 @@
+        sync: bool = True,
+        no_editable: bool = False,
+        no_self: bool = False,
+        prerelease: bool | None = None,
+        fail_fast: bool = False,
+        hooks: HookManager | None = None,
+        allow_transitives : bool = False
+    ) -> None:
+        """Update specified packages or all packages"""
+        from itertools import chain
+
+        from pdm.cli.actions import do_lock, do_sync
+        from pdm.cli.utils import check_project_file, populate_requirement_names, save_version_specifiers
+        from pdm.models.requirements import strip_extras
+        from pdm.models.specifiers import get_specifier
+        from pdm.utils import normalize_name
+
+        hooks = hooks or HookManager(project)
+        check_project_file(project)
+        if len(packages) > 0 and (top or len(selection.groups) > 1 or not selection.default):
+            raise PdmUsageError(
+                "packages argument can't be used together with multiple -G or " "--no-default or --top."
+            )
+        all_dependencies = project.all_dependencies
+        updated_deps: dict[str, dict[str, Requirement]] = defaultdict(dict)
+        locked_groups = project.lockfile.groups
+        if not packages:
+            if prerelease is not None:
+                raise PdmUsageError("--prerelease/--stable must be used with packages given")
+            selection.validate()
+            for group in selection:
+                updated_deps[group] = all_dependencies[group]
+        else:
+-            group = selection.one()
+-            if locked_groups and group not in locked_groups:
+-                raise ProjectError(f"Requested group not in lockfile: {group}")
+-            dependencies = all_dependencies[group]
+-            for name in packages:
+-                normalized_name = normalize_name(name)
+-                matched_req = next(
+-                    (v for k,v in dependencies.items() if normalize_name(strip_extras(k)[0]) == normalized_name),
+-                    None,
+-                )
+-                if not matched_req and allow_transitives:
+-                    candidates = project.locked_repository.all_candidates
+-                    matched_req = next(
+-                        (v.req for k,v in candidates.items() if normalize_name(strip_extras(k)[0]) == normalized_name),
+-                        None,
+-                    )
+-                if not matched_req:
+-                    raise ProjectError(
+-                        f"[req]{name}[/] does not exist in [primary]{group}[/] "
+-                        f"{'dev-' if selection.dev else ''}dependencies."
+-                    )
+-                matched_req.prerelease = prerelease
+-                updated_deps[group][normalized_name] = matched_req
++                mathced_req, updated_deps = update_groups_only(selection, locked_groups, all_dependencies, packages, normalize_name, dependencies, updated_deps, allow_transitives, project)
```

Comment:
Borde vara trärtom, branched `if not packages` är den som updaterar grupper. Eller?